pkg/querier/querier.go

```diff
@@ -1283,11 +1283,16 @@ func getStructuredMetadata(entry push.Entry) map[string][]string {
 	return result
 }
 
-func parseLine(line string) (map[string][]string, *string) {
+func parseLine(line string, stream logproto.Stream) (map[string][]string, *string) {
+	streamLbls, err := syntax.ParseLabels(stream.Labels)
```

Comment:
that's what I had in https://github.com/grafana/loki/pull/13993/commits/c51511016efa3946d98f4fba06acc00b956304f7, but I was worried about the performance hit of having to iterate over the streams twice. I also don't know if this solves the problem, take your example:

line 1: {app="foo"} with stream labels: app -> detected_field will do app_extracted
line 2: {app="foo"} with stream labels: bar -> ~detected_field will do app~ (detected fields will only include bar)

that second case is not true, because [we're filtering out the stream labels from the detected field response](https://github.com/grafana/loki/pull/13993/files#diff-a0b881d1b7b99f439716f189b50eef90ccdaea00845b087c0dbce0e148aa2c0eR1310). Furthermore, doing a `sum by (app_extracted)` shouldn't return results for line 2, right? which is what we want?

So, wouldn't this produce the following, correct result?

`sum by (app_extracted) (count_over_time({app="foo"}))` -> 1
`sum by (app) (count_over_time({app="foo"}))` -> 2

Reply:
Ok, tested this in Ops. I think we want to return both `app` and `app_extracted` given that scenario, which is what the code should do in it's current state.

Here's the test, given the query `sum by (app) (count_over_time({service_name=~"(grafana/grafana|loki/querier)"} | line_format "app=foo" | logfmt[$__auto]))`, where `grafana/grafana` has an indexed `app` label and `loki/querier` does not, the [result is 2 series](https://ops.grafana-ops.net/explore?schemaVersion=1&panes=%7B%227ne%22:%7B%22datasource%22:%22OP27Xzxnk%22,%22queries%22:%5B%7B%22refId%22:%22B%22,%22expr%22:%22sum%20by%20%28app%29%20%28count_over_time%28%7Bservice_name%3D~%5C%22%28grafana%2Fgrafana%7Cloki%2Fquerier%29%5C%22%7D%20%7C%20line_format%20%5C%22app%3Dfoo%5C%22%20%7C%20logfmt%5B$__auto%5D%29%29%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22OP27Xzxnk%22%7D,%22editorMode%22:%22code%22,%22hide%22:false%7D,%7B%22refId%22:%22E%22,%22expr%22:%22%7Bservice_name%3D%5C%22loki%2Fquerier%5C%22%7D%20%7C%20logfmt%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22OP27Xzxnk%22%7D,%22editorMode%22:%22code%22%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D%7D&orgId=1), `app=grafana` and `app=foo`.

There is now way to query saying, `sum by (app)` but only when it is a non-indexed field, so I think the detected fields logic just needs to be all detected fields _by stream_